### DISCUSSION TITLE
Review Comment API missing side/start-line

### DISCUSSION THREAD

**Discussion by zimventures** [community/community#86055] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

When making a call to [list the comments for a pull review](https://docs.github.com/en/rest/pulls/reviews?apiVersion=2022-11-28#list-comments-for-a-pull-request-review), I don't appear to be seeing any of the details for the side or start/end lines for a multi-line comment. 

For example - here is a review comment made on the right side of a PR:
![image](https://github.com/community/community/assets/23481310/bcfe013f-5cce-488e-8fe2-93e142c762a5)

When performing the API call to get the comments for the review, all I get back is:
```json
[
    {
        "id": 1446295484,
        "url": "[redacted]",
        "pull_request_review_id": 1811492167,
        "diff_hunk": "@@ -9,6 +9,10 @@\n \n This module contains all of the code to execute Google Cloud Functions. Each subdirectory contains a single function that will be deployed within the GCloud. During local development, each function will run as a standalone Flask application. The Flask applications will all run on different ports.\n \n+# This is nothing\n+ignore this text\n+Just looking to do a multi-line change\n+",
        "path": "functions/README.md",
        "position": 7,
        "original_position": 7,
        "commit_id": "a22506f11a44a3baa09f11874d170676d97d2a08",
        "user": {
            [redacted]
        },
        "body": "This is a multi-line comment on the right side to talk about something.",
        "created_at": "2024-01-09T16:08:42Z",
        "updated_at": "2024-01-09T16:09:00Z",
        "html_url": "[redacted]",
        "pull_request_url": "[redacted]",
        "author_association": "CONTRIBUTOR",
        "_links": {
           [redacted]
        },
        "original_commit_id": "a22506f11a44a3baa09f11874d170676d97d2a08",
        "reactions": {
           [redacted]
        }
    }
]
```

Even though the comment spans 3 lines (12-15), the only offset information received is a single relative location in the diff hunk - excluding any notion that the comment was made on multiple lines.

I feel like this is a bug because the API docs mention the presence of `side`, `start_side`, `line`, `original_line`, `start_line`, and `original_start_line` , none of which appear to be present in the results from the API call.

**Comment by Naimulnobel** [comment]
The fields `side`, `start_side`, `line`, `original_line`, `start_line`, and `original_start_line` are indeed mentioned in the GitHub API documentation, but they are only included in the response when the comment is part of a review that was submitted with the "Review changes" option in the GitHub UI. 

If the comment was added directly in the "Conversation" tab of the pull request, or if it was added as a single comment (not part of a review), these fields will not be included in the API response.

If you're not seeing these fields in the API response for comments that are part of a review, it might be a bug or limitation in the GitHub API. In this case, I would recommend reaching out to GitHub Support for further assistance.

As a workaround, you might be able to infer the lines that the comment applies to from the `diff_hunk` field, which contains the lines of code that were changed in the diff. However, this would require parsing the diff hunk, which can be complex and might not always be accurate.
